File: Data Structures & Algorithms/counting-bits/submission-1.cpp
class Solution {
public:
    vector<int> countBits(int n) {
        vector<int> res;
        for(int i = 0; i<=n ; i++){
            int count = 0;
            int temp = i;
            while(!(temp==0 ||  temp==1)){
                if(temp%2!=0){
                    count ++;
                }
                temp/=2;
            }
            if (temp == 1) count ++;
            res.push_back(count);
        }
        return res;
    }
};

File: Data Structures & Algorithms/partition-labels/submission-1.py
class Solution:
    def partitionLabels(self, s: str) -> List[int]:
        lastc = {}
        res = []
        for i, c in enumerate(s):
            lastc[c] = i
        count = 0
        end = 0
        for i , c in enumerate(s):
            end = max(end, lastc[c])
            count +=1
            if end == i:
                res.append(count)
                count = 0
        return res
                